src/cargo/core/shell.rs

```diff
@@ -122,49 +142,26 @@ impl MultiShell {
 
 impl Shell {
     pub fn create(out: Box<Write + Send>, config: ShellConfig) -> Shell {
-        if config.tty && config.color {
-            let term = TerminfoTerminal::new(out);
-            term.map(|t| Shell {
-                terminal: Colored(Box::new(t)),
-                config: config
-            }).unwrap_or_else(|| {
-                Shell { terminal: NoColor(Box::new(io::stderr())), config: config }
-            })
-        } else {
-            Shell { terminal: NoColor(out), config: config }
-        }
+        let term = TerminfoTerminal::new(out);
+        term.map(|t| Shell {
+            terminal: Colored(Box::new(t)),
+            config: config
+        }).unwrap_or_else(|| {
+            Shell { terminal: NoColor(Box::new(io::stderr())), config: config }
+        })
     }
 
-    pub fn verbose<F>(&mut self, mut callback: F) -> io::Result<()>
-        where F: FnMut(&mut Shell) -> io::Result<()>
-    {
-        match self.config.verbosity {
-            Verbose => return callback(self),
-            _ => Ok(())
-        }
-    }
-
-    pub fn concise<F>(&mut self, mut callback: F) -> io::Result<()>
-        where F: FnMut(&mut Shell) -> io::Result<()>
-    {
-        match self.config.verbosity {
-            Verbose => Ok(()),
-            _ => return callback(self)
-        }
+    pub fn set_color_config(&mut self, color_config: ColorConfig) {
+        self.config.color_config = color_config;
     }
 
     pub fn say<T: ToString>(&mut self, message: T, color: Color) -> io::Result<()> {
-        match self.config.verbosity {
-            Quiet => Ok(()),
-            _ => {
-                try!(self.reset());
-                if color != BLACK { try!(self.fg(color)); }
-                try!(write!(self, "{}\n", message.to_string()));
-                try!(self.reset());
-                try!(self.flush());
-                Ok(())
-            },
-        }
```

Comment:
Same here. No verbosity so the match was removed and the contents of the `_` arm were left. This was the original functionality.